### PR TITLE
Use command FSM from commons plugin & Enable AppInsight

### PR DESCRIPTION
**Pull request by ArchangelSDY**
This PR is based on #5 and #7 and includes following changes:

* Use command FSM from commons plugin
* Enable AppInsight

### CODE REVIEW

File: src/main/java/com/microsoft/jenkins/appservice/commands/DockerDeployCommand.java [hunk 1]
```diff
@@ -85,11 +91,19 @@ public void execute(final IDockerDeployCommandData context) {
                 azure.webApps().inner().updateConfigurationSlot(
                         webApp.resourceGroupName(), webApp.name(), slot.name(), siteConfigResourceInner);
             }
-            context.setDeploymentState(DeploymentState.Success);
+            context.setCommandState(CommandState.Success);
             context.logStatus("Azure app service updated successfully.");
+            AzureAppServicePlugin.sendEvent(Constants.AI_WEB_APP, Constants.AI_DOCKER_DEPLOY,
+                    "WebApp", AppInsightsUtils.hash(context.getWebApp().name()),
```
Comment (chenkennt) [inline src/main/java/com/microsoft/jenkins/appservice/commands/DockerDeployCommand.java]: Also include subscription ID and resource group name

Comment (ArchangelSDY) [reply src/main/java/com/microsoft/jenkins/appservice/commands/DockerDeployCommand.java]: Added resource group name. Subscription ID is included by default inside the AppInsight SDK.

Comment (ArchangelSDY) [reply src/main/java/com/microsoft/jenkins/appservice/commands/DockerDeployCommand.java]: Sorry, misunderstood it. Will add subscription ID and job ID soon.

File: src/main/java/com/microsoft/jenkins/appservice/WebAppDeploymentCommandContext.java [hunk 2]
```diff
@@ -109,38 +114,40 @@ public void configure(
             pubProfile = slot.getPublishingProfile();
         }
 
-        HashMap<Class, TransitionInfo> commands = new HashMap<>();
+        final AzureCredentials.ServicePrincipal sp = AzureCredentials.getServicePrincipal(azureCredentialsId);
+
+        AzureAppServicePlugin.sendEvent(Constants.AI_WEB_APP, Constants.AI_START_DEPLOY,
+                "Run", AppInsightsUtils.hash(run.getUrl()),
```
Comment (chenkennt) [inline src/main/java/com/microsoft/jenkins/appservice/WebAppDeploymentCommandContext.java]: One suggestion is to add this as common parameter of sendEvent.

**Pull request merged**